### PR TITLE
save load path in state and pass to mosaicmllogger

### PR DESCRIPTION
**Pull request by ethanma-db**
# What does this PR do?

Goal is to provide more visibility in the run life cycle. To do this, we can indicate when a run is resumed from checkpoint or not by displaying the load path if it was resumed from a checkpoint. Corresponding mcloud PR with `mcli describe` changes:
https://github.com/databricks-mosaic/mcloud/pull/4338
# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Review by dakinggg — COMMENTED** [review]
please add a pr description if you're ready for review

Edit: you did :)

**Review by KuuCi — APPROVED** [review]
This looks good to me!

File: composer/loggers/mosaicml_logger.py [hunk 1]
```diff
@@ -98,6 +98,11 @@ def log_metrics(self, metrics: dict[str, Any], step: Optional[int] = None) -> No
         self.log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # If model was resumed from checkpoint, log the checkpoint path
+        if int(state.timestamp.batch) > 0 and state.load_path is not None:
+            log.debug(f'Logging checkpoint path to metadata: ', state.load_path)
```
Comment (mvpatel2000) [inline composer/loggers/mosaicml_logger.py:103]:
```suggestion
            log.debug(f'Logging checkpoint path to metadata: {state.load_path}')
```

**Review by mvpatel2000 — APPROVED** [review]
LGTM! Left a minor nit

**Review by b-chu — COMMENTED** [review]
Thanks for the PR! I'm hesitant about adding more public attributes to state. Is there something that already exists in state that we can do a before_load and after_load check on @eracah ?

**Comment by mvpatel2000** [comment]
> Thanks for the PR! I'm hesitant about adding more public attributes to state. Is there something that already exists in state that we can do a before_load and after_load check on @eracah ?

Ethan and I discussed offline -- unfortunately i don' t think so because of the autoresume behavior... but open to ideas

**Comment by b-chu** [comment]
No idea unfortunately. I'm not blocking so feel free to merge

**Pull request merged**